src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -446,6 +447,9 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         }
     }
 
+    if (joined_tables.tablesCount() > 1 && settings.allow_experimental_parallel_reading_from_replicas)
```

Comment:
Not a good solution. It's better to automatically disable parallel replicas for JOIN. Or we couldn't enable it by default.